index.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    arr = to_a
+    arr.to_enum unless block_given?
+    n = 0
+    while arr.length > n
+      yield(arr[n])
+      n += 1
+    end
+  end
+
+  def my_each_with_index
+    arr = to_a
+    return arr.to_enum unless block_given?
+
+    n = 0
+    while arr.length > n
+      yield(arr[n], n)
+      n += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    res = []
+    my_each do |v|
+      res << v if yield(v) == true
+    end
+    res
+  end
+
+  def my_all?
+    check = true
+    if block_given?
+      my_each { |i| check = false unless yield(i) }
+    else
+      my_all? { |obj| obj }
+    end
+    check
+  end
+
+  def my_any?
+    check = false
+    if block_given?
+      my_each { |i| check = true if yield(i) }
+    else
+      my_any? { |obj| obj }
+    end
+    check
+  end
+
+  def my_none?
+    check = true
+    if block_given?
+      my_each { |i| check = false if yield(i) }
```

Comment:
- [ ] my_none? when no block or argument is given returns true only if none of the collection members is true
- [ ] my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
- [ ] my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex
- [ ] my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern